src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class WebSocketsTransport : ITransport
+    {
+        private ClientWebSocket _webSocket = new ClientWebSocket();
+        private IChannelConnection<Message> _application;
+        private CancellationToken _cancellationToken = new CancellationToken();
+        private readonly ILogger _logger;
+
+        public WebSocketsTransport()
+        {
+        }
+        public WebSocketsTransport(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<WebSocketsTransport>();
+        }
+        public Task Running { get; private set; }
+                                                
+        public async Task StartAsync(Uri url, IChannelConnection<Message> application)  
+        {
+            if (url == null)
+            {
+                throw new ArgumentNullException(nameof(url));
+            }
+            if (application == null)
+            {
+                throw new ArgumentNullException(nameof(application));
+            }
+            _application = application;
+            await Connect(url);
+            var sendTask = SendMessages(url, _cancellationToken);
+            var receiveTask = ReceiveMessages(url, _cancellationToken);
+
+            Running = Task.WhenAll(sendTask, receiveTask).ContinueWith(t => {
```

Comment:
I would actually say iterate. Converting this to the new pattern is not a lot of work so it does not make sense to block on this. I am planning to send a PR today but it is quite a big change so may have a lot of feedback to address.